sylt-compiler/src/typechecker.rs

```diff
@@ -1144,19 +1143,6 @@ mod op {
         Type::Tuple(a.iter().map(f).collect())
     }
 
-    fn union_un_op(a: &BTreeSet<Type>, f: fn(&Type) -> Type) -> Type {
-        a.iter()
-            .find_map(|x| {
-                let x = f(x);
-                if x.is_nil() {
-                    None
-                } else {
-                    Some(x)
-                }
-            })
-            .unwrap_or(Type::Invalid)
-    }
```

Comment:
This would still be valid when the un_op is `-` and the type is `int | float`. But I guess we would have to specify that somehow?

Reply:
That's a very good point. It feels like a different PR.